### PR TITLE
Print a binary tree

### PR DESCRIPTION
**Pull request by znichola**
## Theo reformatting
Some small code refactor adding static to functions not used elsewhere, and make file changes.

### tree_ops.c
`get_tree_height` calculates how many level to the tree there are
` get_tree_width` width for a given level in the tree
`get_tree_max_width` maximum width for the whole tree

### print_tree.c
`print_tree` used to recursively print the tree
`auto_print_tree` to do a rough calculation of how much space is needed before printing the tree.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,4 @@
 a.out
+eval
+eval.dSYM
+.vscode
```
Comment (t-h2o) [inline .gitignore]: Where is the `newline` at the end of file !?

File: eval.h [hunk 2]
```diff
@@ -80,6 +88,13 @@ void	parse_factor(char **str, t_tree **tree);
...
+// print_tree.c
+void	print_tree(t_pos p, t_tree *tree);
+void	auto_print_tree(t_tree *tree);
 
+// tree_ops.c
+int		get_tree_width(t_tree *tree, int level);
+int		get_tree_height(t_tree *tree);
+int		get_tree_max_width(t_tree *tree);
 
 #endif
```
Comment (t-h2o) [inline eval.h]: nl

File: makefile [hunk 3]
```diff
@@ -1,16 +1,18 @@
 CC=gcc
-CFLAGS= -Wall -Wextra
+# CFLAGS= -Wall -Wextra -Werror
```
Comment (t-h2o) [inline makefile:2]: why ?

File: print_tree.c [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+{
+	int	width;
+	int	height;
+
+	width = get_tree_max_width(tree) * 6;
+	height = get_tree_height(tree) * 4;
+	print_tree((t_pos){width, 2}, tree);
+	printf("\033[%d;0H", height);
+	printf("\n");
+}
```
Comment (t-h2o) [inline print_tree.c]: nl ?

**Review by t-h2o — APPROVED** [review]

File: makefile (hunk 3 above)
Comment (znichola) [reply makefile:2]: So you have something to comment! But I accept.

File: print_tree.c (hunk 4 above)
Comment (znichola) [reply print_tree.c]: done: 6008e72defbbdd2ca91543fbeccf417aa81b7698

File: eval.h (hunk 2 above)
Comment (znichola) [reply eval.h]: done: 6008e72defbbdd2ca91543fbeccf417aa81b7698

**Pull request merged**